src/pybind/mgr/nfs/cluster.py

```diff
@@ -206,54 +212,58 @@ def show_nfs_cluster_info(self, cluster_id: Optional[str] = None) -> Tuple[int,
+    def get_nfs_cluster_config(self, cluster_id: str) -> str:
         try:
             if cluster_id in available_clusters(self.mgr):
                 rados_obj = self._rados(cluster_id)
                 conf = rados_obj.read_obj(user_conf_obj_name(cluster_id))
-                return 0, conf or "", ""
+                return conf or ""
             raise ClusterNotFound()
         except Exception as e:
-            return exception_handler(e, f"Fetching NFS-Ganesha Config failed for {cluster_id}")
+            log.exception(f"Fetching NFS-Ganesha Config failed for {cluster_id}")
+            raise ErrorResponse.wrap(e)
 
-    def set_nfs_cluster_config(self, cluster_id: str, nfs_config: str) -> Tuple[int, str, str]:
+    def set_nfs_cluster_config(self, cluster_id: str, nfs_config: str) -> None:
         try:
             if cluster_id in available_clusters(self.mgr):
                 rados_obj = self._rados(cluster_id)
                 if rados_obj.check_user_config():
-                    return 0, "", "NFS-Ganesha User Config already exists"
+                    raise ErrorResponse("NFS-Ganesha User Config already exists",
+                                        return_value=-errno.EEXIST)
                 rados_obj.write_obj(nfs_config, user_conf_obj_name(cluster_id),
                                     conf_obj_name(cluster_id))
                 log.debug("Successfully saved %s's user config: \n %s", cluster_id, nfs_config)
                 restart_nfs_service(self.mgr, cluster_id)
-                return 0, "NFS-Ganesha Config Set Successfully", ""
+                return
             raise ClusterNotFound()
         except NotImplementedError:
-            return 0, "NFS-Ganesha Config Added Successfully "\
-                "(Manual Restart of NFS PODS required)", ""
+            raise ManualRestartRequired("NFS-Ganesha Config Added Successfully")
         except Exception as e:
-            return exception_handler(e, f"Setting NFS-Ganesha Config failed for {cluster_id}")
+            log.exception(f"Setting NFS-Ganesha Config failed for {cluster_id}")
+            raise ErrorResponse.wrap(e)
 
-    def reset_nfs_cluster_config(self, cluster_id: str) -> Tuple[int, str, str]:
+    def reset_nfs_cluster_config(self, cluster_id: str) -> None:
         try:
             if cluster_id in available_clusters(self.mgr):
                 rados_obj = self._rados(cluster_id)
                 if not rados_obj.check_user_config():
-                    return 0, "", "NFS-Ganesha User Config does not exist"
+                    raise ErrorResponse("NFS-Ganesha User Config does not exist",
+                                        return_value=-errno.ENOENT)
                 rados_obj.remove_obj(user_conf_obj_name(cluster_id),
                                      conf_obj_name(cluster_id))
                 restart_nfs_service(self.mgr, cluster_id)
-                return 0, "NFS-Ganesha Config Reset Successfully", ""
+                return
```

Comment:
please, can you shed some light on how this success messages are being delivered to the caller with the new changes?

Reply:
I should have called this out, but in this case it's an intentional behavior change to be consistent with work like PR #45843

Reply:
so I understand we don't show a message in case of success anymore, right?

Reply:
Yup. In the case of success with no information to report, silence is golden. :-)